core/cache/cache.go

```diff
@@ -28,6 +28,8 @@ type (
 		Purge(key string) error
 		PurgeTags(tags []string) error
 		Flush() error
+		TagSupport() bool
```

Comment:
> Semantically you mean "PurgeByTagsSupport" - right?
i don't know why the PurgeTags was an part of the interface before, because no existing cache-backend implements it. but i see more usages for tags-support. f.e GetByTag (which can usage redis MGET-feature).
since it is not use currently, i would suggest to remove the PurgeTags completly.
For the Flush i would suggest to implement it in all backends (only file backend missed). should be a simple task.

> 
> Another option would be to make the two methods above Flush and PurgeTags not part of the main backend interface but add additional interfaces that we can type assert on the places where we want to check for them.
> We use this way of "adding" optional behavior to interfaces on other places as well
Sounds nice. i'll try.

Reply:
+1 on type assertions. Do not change an existing, exposed interface (backwards incompatible)